src/keystore.c

```diff
@@ -427,6 +427,23 @@ static bool _get_xprv(const uint32_t* keypath, const size_t keypath_len, struct
     return true;
 }
 
+bool keystore_get_root_fingerprint(uint8_t* fingerprint)
+{
+    if (sizeof(fingerprint) < sizeof(uint32_t)) {
+        Abort("fingerprint buffer needs to be size 4");
+    }
+    static const uint32_t keypath[] = {
+        0 + BIP32_INITIAL_HARDENED_CHILD,
+    };
+
+    struct ext_key derived_xpub __attribute__((__cleanup__(keystore_zero_xkey))) = {0};
+    if (!keystore_get_xpub(keypath, sizeof(keypath) / sizeof(uint32_t), &derived_xpub)) {
+        return false;
+    }
+    memcpy(fingerprint, derived_xpub.parent160, sizeof(uint32_t));
```

Comment:
I see you take the parent160 here, but it's more straight forward to get the xpub at `m/` and copy from `derived_xpub.hash160` instead, which saves some work and should be the same result.

Reply:
Mmh, I disagree here, since it is the hash of the parent public key that is included in the xpub, so getting the fingerprint of a public key always implies taking to the first four bytes of the hash160 of its parent. Further, I don't know how to get from `m/` with the usual keypath syntax :P

Reply:
Hmm there shouldn't be anyting to disagree on, you have ```fingerprint(root xpub) == parentfingerprint(child xpub)```, so deriving a child to get the parent fingerprint when you already have the parent fingerprint in memory is both physical overhead and mental overhead when reading/maintaining the code :P

> Further, I don't know how to get from m/ with the usual keypath syntax :P

`keystore_get_xpub(NULL, 0, ...)`. Could add a small comment that this is allowed in the docstring of `keystore_get_xpub()`.